lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -437,29 +858,69 @@ void HdVP2Mesh::Sync(
 #endif
     }
 
-    if (HdChangeTracker::IsPrimvarDirty(*dirtyBits, id, HdTokens->normals)
+    if (HdChangeTracker::IsPrimvarDirty(*dirtyBits, id, HdTokens->points)
+        || HdChangeTracker::IsPrimvarDirty(*dirtyBits, id, HdTokens->normals)
         || HdChangeTracker::IsPrimvarDirty(*dirtyBits, id, HdTokens->primvar)) {
-        const HdVP2Material* material = static_cast<const HdVP2Material*>(
-            renderIndex.GetSprim(HdPrimTypeTokens->material, GetMaterialId()));
 
-        const TfTokenVector& requiredPrimvars = material && material->GetSurfaceShader()
-            ? material->GetRequiredPrimvars()
-            : sFallbackShaderPrimvars;
+        auto addRequiredPrimvars = [&](const SdfPath& materialId) {
+            const HdVP2Material* material = static_cast<const HdVP2Material*>(
+                renderIndex.GetSprim(HdPrimTypeTokens->material, materialId));
+            const TfTokenVector& requiredPrimvars = material && material->GetSurfaceShader()
+                ? material->GetRequiredPrimvars()
+                : sFallbackShaderPrimvars;
+
+            for (const auto& requiredPrimvar : requiredPrimvars) {
+                if (!_PrimvarIsRequired(requiredPrimvar)) {
+                    _meshSharedData->_allRequiredPrimvars.push_back(requiredPrimvar);
+                }
+            }
+        };
+
+        // there is a chance that the geom subsets cover all the faces of the
+        // mesh and that the overall material id is unused. I don't figure that
+        // out until much later, so for now just accept that we might pull unnecessary
+        // primvars required by the overall material but not by any of the geom subset
+        // materials.
+        addRequiredPrimvars(GetMaterialId());
```

Comment:
Shouldn't there be a loop to ask each material of the geom subset for the required primvars?